dev/bots/test.dart

```diff
@@ -75,6 +75,9 @@ int get webShardCount => Platform.environment.containsKey('WEB_SHARD_COUNT')
   ? int.parse(Platform.environment['WEB_SHARD_COUNT'])
   : 8;
 
+/// The number of shards the long-running Web tests are split into.
+const int kWebLongRunningTestShardCount = 3;
```

Comment:
Shard count is not passed to the test script, only the shard index. E.g. when you run it as `SHARD=web_tests SUBSHARD=4 dart dev/bots/test.dart` the script needs to know if it's 4 out of 10, or 4 out of 20. So total shard counts are hard-coded in the test script. See also `kBuildTestShardCount`, `webShardCount`, and usages of `selectSubshard`.

Reply:
It may be because, the shard count is not passed that way to builders.

I usually refer to already running builders to understand how properties work, for example: https://ci.chromium.org/p/flutter/builders/prod/Linux%20web_tests/878

Shards is:  "web_tests", subshards is: [ "0", "1", "2", "3", "4", "5", "6", "7_last" ]

When these are provided is there an information not set by the recipe?

Recipe code is here btw: https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/flutter/flutter_drone.py

Reply:
> When these are provided is there an information not set by the recipe?

Yes, the recipe only sets `SHARD` (e.g. `web_tests`) and `SUBSHARD` (e.g. `3`) environment variables: https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/flutter/flutter_drone.py#55. The recipe does not pass the total number of available shards. That's the missing information.